06_Userspace/02_Switching_Modes.md

```diff
@@ -51,6 +51,44 @@ As for the other two values? We're going to set `rip` to the instruction we want
 
 Since we have paging enabled, that means page-level protections are in effect. If we try to run code from a page that has the NX-bit set (bit 63), we'll page fault. The same is true for trying to run code or access a stack from a page with the U/S bit cleared. On `x86_64` this bit must be set at every level in the paging structure.
 
+This also cause us to face another problem: how to test if userspace is working correctly? If the scheduler has been implemented using the part 5 of this book, just creating a thread with user level `ss` and `cs` is not enough, since the thread to run use the code that is present in the higher half (even the function to execute), and this mean that according to our design that area is marked as supervisor only.
```

Comment:
I think this section might do well under a heading of `Testing userspace`, since its a change of topic